flytekit/tools/translator.py

```diff
@@ -461,8 +461,9 @@ def get_serializable_node(
     elif isinstance(entity.flyte_entity, PythonTask):
         # handle pod template overrides
         override_pod_spec = {}
-        if entity._pod_template is not None and settings.should_fast_serialize():
-            entity.flyte_entity.set_command_fn(_fast_serialize_command_fn(settings, entity.flyte_entity))
+        if entity._pod_template is not None:
+            if settings.should_fast_serialize():
+                entity.flyte_entity.set_command_fn(_fast_serialize_command_fn(settings, entity.flyte_entity))
             override_pod_spec = _serialize_pod_spec(
                 entity._pod_template, entity.flyte_entity._get_container(settings), settings
             )
```

Comment:
It would be nice to add a test to https://github.com/flyteorg/flytekit/blob/master/tests/flytekit/unit/test_translator.py that would have caught this :)

Reply:
> It would be nice to add a test to https://github.com/flyteorg/flytekit/blob/master/tests/flytekit/unit/test_translator.py that would have caught this :)

I'll take a look later today about adding one.